src/components/SidebarContent.vue

```diff
@@ -215,27 +218,32 @@ export default {
       this.page = page
       this.searchAlgolia(this.filters, this.searchInput, this.numberPerPage, this.page)
     },
-    searchSciCrunch: function(params) {
-      this.loadingScicrunch = true
-      this.scrollToTop();
-      this.$emit("search-changed", { value: this.searchInput, type: "query-update" });
-      this.callSciCrunch(this.envVars.API_LOCATION, params)
-        .then(result => {
-          //Only process if the search term is the same as the last search term.
-          //This avoid old search being displayed.
-          this.sciCrunchError = false;
-          this.resultsProcessing(result);
-          this.$refs.content.style["overflow-y"] = "scroll";
-          this.loadingCards = false;
-          this.loadingScicrunch = false
-        })
-        .catch(result => {
-          if (result.name !== 'AbortError') {
-            this.loadingCards = false;
-            this.loadingScicrunch = false
-            this.sciCrunchError = result.message;
-          }
-        })
+    handleMissingData: function(doi) {
+      let i = this.results.findIndex(res=> res.doi === doi)
+      if (this.results[i])
+        this.results[i].detailsReady = true;
+    },
+    perItemSearch: function() {
+      const doi = this._dois.shift();
```

Comment:
`perItemSearch` in this form is making one request at a time for the cards.  Is there a reason not to do them in parallel?

Reply:
In theory it can and it should but in reality the current API is on http/1.1 and that does not works well with parallel requests at this moment, I am doing this to avoid request timeout.

Reply:
I will do some testing on that and check if I can get much performance boost from it.

Reply:
> current API is on http/1.1 and that does not works well with parallel requests
oh interesting, did we get timeouts before?

Reply:
Occasionally. I have tested running them in parallel and it seems to work alright. I will check if it works well when I change the page with all those requests with abort controller.

Reply:
I do think running in parallel is faster, I am going to get it to run in parallel. Thanks.